### PR TITLE
47 create export algorithm

### PR DESCRIPTION
**Pull request by MPloier**

### CODE REVIEW

File: components/export_table/ExportTableButton.vue [hunk 1]
```diff
@@ -1,8 +1,98 @@
...
+const items: DropdownItem[][] = []
+
+supportedFileFormate.forEach((format, index) => {
+  items.push([
+    {
+      label: format,
+      shortcuts: ["" + (index + 1)],
+      disabled: (index == 0),
+      icon: (index == 0) ? "i-heroicons-check-16-solid" : "",
+      class: "opacity-100",
```
Comment (Dino-Kupinic) [inline components/export_table/ExportTableButton.vue:23]: this causes an type error
I think we can remove it, or do we really need it?
I tried removing it and nothing changed

File: components/export_table/ExportTableButton.vue [hunk 2]
```diff
@@ -1,8 +1,98 @@
...
+  } else if (selectedItem.label == "JSON") {
+    const blob = new Blob([JSON.stringify(teachers, null, 2)], {
+      type: "application/json",
+    })
+
+    return new File([blob], "untisPlanner.json", {
+      type: "application/json",
+    })
+  } else return null
+}
```
Comment (Dino-Kupinic) [inline components/export_table/ExportTableButton.vue:62]: I refactored this to a switch for easier readability

File: components/export_table/ExportTableButton.vue [hunk 3]
```diff
@@ -1,8 +1,98 @@
...
+  const replacer = (key: any, value: any) => value ?? ""
+  const rowItems = items.map((row: any) =>
+      header
+          .map((fieldName) => JSON.stringify(row[fieldName], replacer))
+          .join(","),
+  )
+  // join header and body, and break into separate lines
+  const csv = [headerString, ...rowItems].join("\r\n")
+  return csv
+}
```
Comment (Dino-Kupinic) [inline components/export_table/ExportTableButton.vue:77]: instead of `any`, can we add types?

**Review by Dino-Kupinic — CHANGES_REQUESTED** [review]

**Comment by Dino-Kupinic** [comment]
Please add tests, it is quite annoying testing correct content of the files

File: components/export_table/ExportTableButton.vue (hunk 3 above)
Comment (MPloier) [reply components/export_table/ExportTableButton.vue:77]: Is a function I found and it work --> never change a running system

File: components/export_table/ExportTableButton.vue (hunk 1 above)
Comment (MPloier) [reply components/export_table/ExportTableButton.vue:23]: Everything is fine

File: components/export_table/ExportTableButton.vue (hunk 2 above)
Comment (MPloier) [reply components/export_table/ExportTableButton.vue:62]: okay

**Pull request closed without merging**